src/pykeen/nn/node_piece.py

```diff
@@ -272,7 +296,11 @@ def extra_repr(self) -> Iterable[str]:  # noqa: D102
         yield f"selections={self.selections}"
 
     def __call__(self, edge_index: numpy.ndarray) -> numpy.ndarray:  # noqa: D102
-        return numpy.concatenate([selection(edge_index=edge_index) for selection in self.selections])
+        anchors = numpy.array([], dtype=numpy.int64)
+        for selection in self.selections:
+            anchors = selection(edge_index=edge_index, known_anchors=anchors)
```

Comment:
since the anchors are now selected sequentially instead of independently, the order of selection heuristics may now matter. Maybe we should document this somewhere.

Reply:
makes sense, added in https://github.com/pykeen/pykeen/pull/776/commits/b7fc453318e0b5dceb52c8b2c8f7c071a58a44e4